lib/stubhub/client.rb

```diff
@@ -53,65 +53,71 @@ def login(opts = {})
     end
 
     def create_listing(opts = {})
-      url = "/inventory/listings/v1"
-
-
-      listing_params = {
+      url = "/inventory/listings/v2"
+      
+      listing = {
           eventId: opts[:event_id],
           quantity: opts[:quantity],
-          seats: opts[:seats],
-          pricePerTicket: {
-            amount: opts[:price_per_ticket].to_s,
+          pricePerProduct: {
+            amount: opts[:price_per_ticket].to_f.round(2),
             currency: 'USD'
           },
           deliveryOption: opts[:delivery_option],
-          ticketTraits: [],
-          rows: opts[:rows],
           section: opts[:section],
+          products: [],
+          ticketTraits: [],
+          externalListingId: opts[:external_id]
           # status: 'INACTIVE' # Disable for production
       }
 
       opts[:traits].each do |trait|
-        listing_params[:ticketTraits].push({id: trait.to_s})
+        listing[:ticketTraits].push({id: trait.to_s,operation: "ADD"})
       end
-
+      puts listing
+      # products 
+      if opts[:rows].count == 1
+        opts[:seats].each do |seat|
+          listing[:products].push({row:opts[:rows][0],
+            productType:"TICKET",seat:seat,operation: "ADD",externalId: opts[:external_id]})
+        end
+      else
+        # for piggyback 
+        length = opts[:seats].count/2
+        opts[:rows].each do |row|
+          0.upto(length-1) do |i|
+            listing[:products].push({row:row,
+              productType:"TICKET",seat:opts[:seats][i],operation: "ADD",externalId: opts[:external_id]})
+          end
+        end
+      end
+      
       if opts[:split_option] == -1
-        listing_params[:splitOption] = 'NOSINGLES'
+        listing[:splitOption] = 'NOSINGLES'
       elsif opts[:split_option] == 0
-        listing_params[:splitOption] = 'NONE'
+        listing[:splitOption] = 'NONE'
       else
-        listing_params[:splitOption] = 'MULTIPLES'
-        listing_params[:splitQuantity] = opts[:split_option]
+        listing[:splitOption] = 'MULTIPLES'
+        listing[:splitQuantity] = opts[:split_option]
       end
 
       if opts[:in_hand_date]
-        listing_params[:inhandDate] = opts[:in_hand_date]
-      end
-
-      if opts.include? :tickets
-         url = "/inventory/listings/v1/barcodes"
-         listing_params[:tickets] = opts[:tickets]
```

Comment:
what about this tickets logic which changes the "v1/barcodes". Is it not needed anymore?